src/mixins/filing-template-mixin.ts

```diff
@@ -171,7 +198,21 @@ export default class FilingTemplateMixin extends Vue {
     this.setNameRequest(filing.incorporationApplication.nameRequest)
 
     // Set Name Translations
-    this.setNameTranslations(filing.incorporationApplication.nameTranslations?.new)
+    if (filing.incorporationApplication.nameTranslations instanceof Array) {
+      // If it's an array that means it's a draft which is saved from edit-ui by staff.
+      this.setNameTranslations(filing.incorporationApplication.nameTranslations)
+    } else {
+      // If it's an object that means it's an initial draft created from filing-ui and has a structure of an IA.
```

Comment:
Thanks for the helpful comments but now I'm confused... Why do we have 2 different possibilities here? And if there are unchanged translations, what do they look like?

Translations may be a special case but it was my understanding that a corrected IA is a complete IA object in a correction body.

Reply:
All unchanged translations will still be in array with action as null (draft).

Reply:
NameTranslationDraftIF[] is used only for draft state. NameTranslationIF is aligned to the schema of IA

Reply:
Can you please post an example of the IA structure with: some new/modified/ceased translations, and some unchanged ones?

Reply:
When staff initiate a correction: `'nameTranslations': {'new': ['ABC Ltd.', 'Société Générale']}`
Then edit ui converts it into` [{'value':'ABC Ltd.', 'oldValue': null, 'action': null},  {'value': 'Société Générale', 'oldValue': null, 'action': null}]` here to keep track of changes in draft mode.
Then while filing this array will be converted to `{'new': [], 'modified': {'newValue': 'XYZ Ltd.', 'oldValue': 'ABC Ltd.'}, 'ceased': ['Société Générale']}`
If staff add any translation it will be in the `new` array

Reply:
Can you please post an example of the IA structure with: some new/modified/ceased translations, and some unchanged ones?

Reply:
`{'new': ['i am new'], 'modified': {'newValue': 'XYZ Ltd.', 'oldValue': 'ABC Ltd.'}, 'ceased': ['Société Générale']}`
unchanged once will not be part of correction filing, hence stripped out while filing

Reply:
OK.
So I think this goes against the idea that a correction includes a COMPLETE IA filing structure.
@lekshmimallika-aot Does this seem right to you? Does the back end retain the unchanged translations?

Reply:
This looks ok to me. Yes backend retain the unchanged translations

Reply:
OK for now